fetch-api/fetch-reddit.html

```diff
@@ -0,0 +1,118 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+
+    <meta name="description" content="Sample illustrating the use of Fetch API.">
+
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+
+    <title>Fetch API Reddit Sample</title>
+
+    <!-- Add to homescreen for Chrome on Android -->
+    <meta name="mobile-web-app-capable" content="yes">
+    <link rel="icon" sizes="192x192" href="../images/touch/chrome-touch-icon-192x192.png">
+
+    <!-- Add to homescreen for Safari on iOS -->
+    <meta name="apple-mobile-web-app-title" content="Fetch API Reddit Sample">
+
+    <meta name="apple-mobile-web-app-capable" content="yes">
+    <meta name="apple-mobile-web-app-status-bar-style" content="black">
+    <link rel="apple-touch-icon-precomposed" href="../images/apple-touch-icon-precomposed.png">
+
+    <!-- Tile icon for Win8 (144x144 + tile color) -->
+    <meta name="msapplication-TileImage" content="images/touch/ms-touch-icon-144x144-precomposed.png">
+    <meta name="msapplication-TileColor" content="#3372DF">
+
+    <link rel="icon" href="../images/favicon.ico">
+
+    <link rel="stylesheet" href="../styles/main.css">
+</head>
+
+<body>
+<h1>Fetch API Reddit Sample</h1>
+
+<p>Available in <a href="http://www.chromestatus.com/feature/6730533392351232">Chrome 42+</a></p>
+
+<p>This is an example of fetching some data (Reddit subreddits) which is then used to
+    populate a menu allowing a user to make another fetch (for the subreddit content).</p>
+
+<div class="output">
+    <p id="menu"></p>
+    <p id="log"></p>
+</div>
+
+<!-- // [START code-block] -->
+<script>
+    "use strict";
+
+    var logEl =  document.querySelector('#log');
+    var menuEl =  document.querySelector('#menu');
+
+    function log() {
+        logEl.innerHTML += Array.prototype.join.call(arguments, '') + '\n';
+    }
+
+    function fetchSubreddit(url) {
+        if(url) {
+            fetch('https://www.reddit.com' + url + '.json')
+                    .then(function(response) {
+                        return response.json()
+                    }).then(function(body) {
+                        var links = "";
+                        for(var i = 0; i < body.data.children.length; i++){
+                            links += '<li><a href="' + body.data.children[i].data.url + '">' + body.data.children[i].data.url + '</a></li>';
+                        }
+                        logEl.innerHTML = '<ul>' + links + '</ul>';
+                    });
+        }
+    }
+
+    fetch('https://www.reddit.com/reddits.json')
+            .then(function(response) {
+                return response.json()
+            }).then(function(json) {
+                console.log(json);
```

Comment:
I intended to keep this to demonstrate the complete payload, but didn't want to log to the page. If it's better to drop these to be consistent with other demos happy to do so.